docs/config.rst

```diff
@@ -553,6 +555,17 @@ Complete list of settings that you can put into ``testenv*`` sections:
     will make tox install all dependencies from this PyPI index server
     (including when installing the project sdist package).
 
+    The recommended way to set custom index server URL is to use :conf:`setenv`:
+
+    .. code-block:: ini
+
+        [testenv]
+        setenv =
+            PIP_INDEX_URL = {env:PIP_INDEX_URL:https://pypi.org/simple/}
+
+    This will ensure the desired index server gets used for virtual environment
+    creation and allow overriding index server URL with environment variable.
```

Comment:
creation and allow overriding the index server URL with an environment variable.